lib/streamlit/elements/arrow.py

```diff
@@ -51,9 +72,112 @@
     None,
 ]
 
+SelectionMode: TypeAlias = Literal[
+    "single-row", "multi-row", "single-column", "multi-column"
+]
+_SELECTION_MODES: Final[set[SelectionMode]] = {
+    "single-row",
+    "multi-row",
+    "single-column",
+    "multi-column",
+}
```

Comment:
curious: is this a typescript thing that we have to define it twice like this?

Reply:
These are a bit different things the type alias is purely for type checking, while _SELECTION_MODES is a value that we can use within the python logic.